lib/matplotlib/ticker.py

```diff
@@ -1906,28 +1921,35 @@ def _staircase(steps):
     def set_params(self, **kwargs):
         """Set parameters within this locator."""
         if 'nbins' in kwargs:
-            self._nbins = kwargs['nbins']
+            self._nbins = kwargs.pop('nbins')
             if self._nbins != 'auto':
                 self._nbins = int(self._nbins)
         if 'symmetric' in kwargs:
-            self._symmetric = kwargs['symmetric']
+            self._symmetric = kwargs.pop('symmetric')
         if 'prune' in kwargs:
-            prune = kwargs['prune']
+            prune = kwargs.pop('prune')
             if prune is not None and prune not in ['upper', 'lower', 'both']:
                 raise ValueError(
                     "prune must be 'upper', 'lower', 'both', or None")
             self._prune = prune
         if 'min_n_ticks' in kwargs:
-            self._min_n_ticks = max(1, kwargs['min_n_ticks'])
+            self._min_n_ticks = max(1, kwargs.pop('min_n_ticks'))
         if 'steps' in kwargs:
-            steps = kwargs['steps']
+            steps = kwargs.pop('steps')
             if steps is None:
                 self._steps = np.array([1, 1.5, 2, 2.5, 3, 4, 5, 6, 8, 10])
             else:
                 self._steps = self._validate_steps(steps)
             self._extended_steps = self._staircase(self._steps)
         if 'integer' in kwargs:
-            self._integer = kwargs['integer']
+            self._integer = kwargs.pop('integer')
+        if kwargs:
+            param = next(k for k in kwargs)
```

Comment:
You can do `k, _ = kwargs.popitem()` or `k, *_ = kwargs` which are both more natural IMO, but it's not a big deal.

Reply:
Will do tomorrow (and also fix the flake8 issue)

Reply:
Comments handled.